File: 2019/day1/Node/part1.js
const { assert } = require('console');
const fs = require('fs');

function part1(filepath) {
    sum = 0;
    fs.readFileSync(filepath, 'UTF8')
        .toString().split('\n')
        .forEach((element,index,arr) => { sum += Math.floor(element/3)-2 });
    return sum
}

assert(part1(__dirname+'/../testcases/test1.txt')==34241)
assert(part1(__dirname+'/../input.txt')==3291356)
console.log("Part 1:",part1(__dirname+'/../input.txt'))

File: 2019/day1/Lua/part1.lua
dofile("UsefulStuff.lua")

local function part1(fileName)
	local total = 0
	for memory in io.lines(Script_path() .. fileName) do
		total = total + math.floor(memory/3)-2
	end
	return total
end

UnitTest(part1, "../testcases/test1.txt", 34241)
UnitTest(part1, "../input.txt", 3291356, "Part 1")
